### PR TITLE
feat: add `operate-schema` command

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

implement #720.

### What is changed and how it works?

add `operate-schema` to support `get`/`set`/`remove` schema of a table.

```
>>> operate-schema get -s=mysql-replica-01 sequence_sharding_optimistic -d sharding_seq_opt -t t1
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "CREATE TABLE `t1` ( `id` bigint(20) NOT NULL, `c2` varchar(20) DEFAULT NULL, `c3` int(11) DEFAULT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_bin",
            "source": "mysql-replica-01",
            "worker": "worker1"
        }
    ]
}


>>> operate-schema set -s=mysql-replica-01 sequence_sharding_optimistic -d sharding_seq_opt -t t1 schema.sql
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "source": "mysql-replica-01",
            "worker": "worker1"
        }
    ]
}

>>> operate-schema remove -s=mysql-replica-01 sequence_sharding_optimistic -d sharding_seq_opt -t t1
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "source": "mysql-replica-01",
            "worker": "worker1"
        }
    ]
}
```

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to update the documentation

### CODE REVIEW

File: syncer/schema.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		stmt.IfNotExists = false // we must ensure drop the previous one.
+
+		var newCreateSQLBuilder strings.Builder
+		restoreCtx := format.NewRestoreCtx(format.DefaultRestoreFlags, &newCreateSQLBuilder)
+		if err = stmt.Restore(restoreCtx); err != nil {
+			return "", terror.ErrSchemaTrackerRestoreStmtFail.Delegate(err)
+		}
+		newSQL := newCreateSQLBuilder.String()
+
+		// drop the previous schema fist.
```
Comment (WangXiangUSTC) [inline syncer/schema.go:64]:
```suggestion
		// drop the previous schema first.
```

File: pkg/schema/tracker.go [hunk 2]
```diff
@@ -92,6 +95,33 @@ func (tr *Tracker) GetTable(db, table string) (*model.TableInfo, error) {
...
 
+// GetCreateTable returns the `CREATE TABLE` statement of the table.
+func (tr *Tracker) GetCreateTable(ctx context.Context, db, table string) (string, error) {
+	name := dbutil.TableName(db, table)
+	// use `SHOW CREATE TABLE` now, another method maybe `executor.ConstructResultOfShowCreateTable`.
+	rs, err := tr.se.Execute(ctx, fmt.Sprintf("SHOW CREATE TABLE %s", name))
+	if err != nil {
+		return "", err
+	}
+	defer rs[0].Close()
```
Comment (WangXiangUSTC) [inline pkg/schema/tracker.go:106]: do we need to judge the `len(rs)`?

File: pkg/schema/tracker.go [hunk 3]
```diff
@@ -92,6 +95,33 @@ func (tr *Tracker) GetTable(db, table string) (*model.TableInfo, error) {
...
+	err = rs[0].Next(ctx, req)
+	if err != nil {
+		return "", err
+	}
+	if req.NumRows() == 0 {
+		return "", nil // this should not happen.
+	}
+
+	row := req.GetRow(0)
+	str := row.GetString(1) // the first column is the table name.
```
Comment (WangXiangUSTC) [inline pkg/schema/tracker.go:118]:
```suggestion
	str := row.GetString(1) // the first column is the table schema.
```

do you mean this?

File: tests/sequence_sharding_optimistic/run.sh [hunk 4]
```diff
@@ -102,6 +137,57 @@ run() {
...
+
+    # try to set an invalid schema.
+    curl -X PUT ${API_URL} -d '{"op":2, "task":"sequence_sharding_optimistic", "sources": ["mysql-replica-01"], "database":"sharding_seq_opt", "table":"t1", "schema":"invalid create table statement"}' > ${WORK_DIR}/get_schema.log > ${WORK_DIR}/set_schema.log
+    check_log_contains ${WORK_DIR}/set_schema.log 'is not a valid `CREATE TABLE` statement' 1
+
+    # try to get schema again, no one exist.
+    curl -X PUT ${API_URL} -d '{"op":1, "task":"sequence_sharding_optimistic", "sources": ["mysql-replica-01"], "database":"sharding_seq_opt", "table":"t1"}' > ${WORK_DIR}/get_schema.log
+    check_log_contains ${WORK_DIR}/get_schema.log "Table 'sharding_seq_opt.t1' doesn't exist" 1
+
+    # try to set another schema, `c3` `int` -> `bigint`.
```
Comment (GMHDBJD) [inline tests/sequence_sharding_optimistic/run.sh:168]: Change it to dmctl command for test?

File: pkg/schema/tracker.go (hunk 3 above)
Comment (csuzhangxc) [reply pkg/schema/tracker.go:118]: No, the first (`0`) is the name, the second(`1`) is the schema.

File: pkg/schema/tracker.go (hunk 2 above)
Comment (csuzhangxc) [reply pkg/schema/tracker.go:108]: added in https://github.com/pingcap/dm/pull/764/commits/0a13645341f4e2da90f015feb01b01cd3ded1f79.

File: tests/sequence_sharding_optimistic/run.sh (hunk 4 above)
Comment (csuzhangxc) [reply tests/sequence_sharding_optimistic/run.sh:168]: changed in https://github.com/pingcap/dm/pull/764/commits/ab9793f6cb8f585d6cf20b43d6ce14f3a1be36ac. PTAL again.

**Comment by csuzhangxc** [comment]
docs PR, https://github.com/pingcap/docs-dm/pull/229

**Pull request merged**